server/start.go

```diff
@@ -338,98 +332,76 @@ func startInProcess(ctx *Context, clientCtx client.Context, appCreator types.App
 		}
 	}
 
-	metrics, err := startTelemetry(config)
+	metrics, err := startTelemetry(srvcfg)
 	if err != nil {
 		return err
 	}
 
-	var apiSrv *api.Server
-	if config.API.Enable {
+	var (
+		apiSrv  *api.Server
+		grpcSrv *grpc.Server
+	)
+
+	if srvcfg.API.Enable {
 		genDoc, err := genDocProvider()
 		if err != nil {
 			return err
 		}
 
 		clientCtx := clientCtx.WithHomeDir(home).WithChainID(genDoc.ChainID)
 
-		apiSrv = api.New(clientCtx, ctx.Logger.With("module", "api-server"))
-		app.RegisterAPIRoutes(apiSrv, config.API)
-		if config.Telemetry.Enabled {
-			apiSrv.SetTelemetry(metrics)
-		}
-		errCh := make(chan error)
-
-		go func() {
-			if err := apiSrv.Start(config); err != nil {
-				errCh <- err
+		if srvcfg.GRPC.Enable {
+			_, port, err := net.SplitHostPort(srvcfg.GRPC.Address)
+			if err != nil {
+				return err
```

Comment:
Is it okay to nest `if srvcfg.GRPC.Enable` block in `if srvcfg.API.Enable`?
`if srvcfg.GRPC.Enable` was previously independent if clause.

Reply:
I will revert that change!